case_studies/galaxy_clustering/prior.py

```diff
@@ -0,0 +1,338 @@
+import pickle
+
+import numpy as np
+import pandas as pd
+from astropy.cosmology.realizations import WMAP7
+from hmf import MassFunction
+from scipy.stats import gennorm
+
+
+class ClusterPrior:
+    def __init__(self):
+        super().__init__()
+        self.size = 100
+        self.width = 5000
+        self.height = 5000
+        self.bands = ["G", "R", "I", "Z"]
+        self.n_bands = 4
+        self.reference_band = 1
+        self.ra_cen = 50.64516228577292
+        self.dec_cen = -40.228830895890404
+        self.mass_min = (10**14.5) * 1.989 * (10**33)  # Minimum value of the range solar mass
+        self.mass_max = (10**15.5) * 1.989 * (10**33)  # Maximum value of the range
+        self.scale_pixels_per_au = 80
+        self.mean_sources = 0.004
+        self.tsize_s = 0.64
+        self.tsize_loc = 0.017
+        self.tsize_scale = 0.23
+        self.mag_ex = 1.3
+        self.mag_max = 25
+        self.bg_min_z = 0
+        self.bg_max_z = 0.3
+        self.cluster_min_z = 0.2
+        self.cluster_max_z = 0.5
+        self.G1_beta = 0.6
+        self.G1_loc = 0
+        self.G1_scale = 0.035
+        self.G2_beta = 0.6
+        self.G2_loc = 0
+        self.G2_scale = 0.032
+        self.tiles_width = 25
+        self.tiles_height = 25
+        with open("gal_gmm_nmgy.pkl", "rb") as f:
+            self.gmm_gal = pickle.load(f)
+        self.tsize_poly = np.poly1d([-6.88890387e-11, 3.70584026e-5, 4.34623392e-2])
+        self.folder_path = "data5/"
+        self.threadings = 8
+        self.redshift_alpha = 1.24
+        self.redshift_beta = 1.01
+        self.redshift0 = 0.51
+        self.cluster_prob = 0.2
+
+    #  Source: https://github.com/LSSTDESC/CLMM/blob/main/clmm/redshift/distributions.py
+    def _redshift_distribution(self, redshift):
+        return (redshift**self.redshift_alpha) * np.exp(
+            -((redshift / self.redshift0) ** self.redshift_beta)
+        )
+
+    def _toflux(self, mag):
+        return 10 ** ((mag - 30) / -2.5)
+
+    def _cluster_threshold(self, r):
+        return 3 / (2 * np.pi * (r**2))
+
+    def _sample_mass(self):
+        hmf = MassFunction()
+        hmf.update(Mmin=14.5, Mmax=15.5)
+        mass_func = hmf.dndlnm
+        mass_sample = []
+        while len(mass_sample) < self.size:
+            index = np.random.randint(0, len(mass_func))
+            prob = (mass_func / sum(mass_func))[index]
+            if np.random.random() < prob:
+                mass_sample.append(
+                    (self.mass_max - self.mass_min) / len(mass_func) * (index + np.random.random())
```

Comment:
I'd be clearer if we used some intermediate variables, rather than have a multi-line append statement